File: ejer_java poo y poli/ejr3.java
package prac1_prog2;

public class ejr3 {
	    public static void main(String[] args) {
	        Producto p1 = new Producto("Arroz", 20, 50);
	        p1.mostrarDatos();
	        p1.vender(10);
	        p1.reabastecer(30);
	        p1.mostrarDatos();
	    }
	}
	class Producto {
	    private String nombre;
	    private double precio;
	    private int stock;

	    public Producto(String nombre, double precio, int stock) {
	        this.nombre = nombre;
	        this.precio = precio;
	        this.stock = stock;
	    }
	    public void vender(int cantidad) {
	        if (cantidad <= stock) {
	            stock -= cantidad;
	            System.out.println("Venta realizada: " + cantidad + " unidades.");
	        } else {
	            System.out.println("Stock insuficiente.");
	        }
	    }
	    public void reabastecer(int cantidad) {
	        stock += cantidad;
	        System.out.println("Se reabasteció con " + cantidad + " unidades.");
	    }
	    public void mostrarDatos() {
	        System.out.println("Producto: " + nombre + " | Precio: " + precio + " | Stock: " + stock);
	    }
	}

File: ejer_phyton poo y poli/ejr3.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= self.stock:
            self.stock -= cantidad
            print(f"Venta realizada: {cantidad} unidades.")
        else:
            print("Stock insuficiente.")

    def reabastecer(self, cantidad):
        self.stock += cantidad
        print(f"Se reabasteció con {cantidad} unidades.")

    def mostrar_datos(self):
        print(f"Producto: {self.nombre} | Precio: {self.precio} | Stock: {self.stock}")

if __name__ == "__main__":
    p1 = Producto("Arroz", 20, 50)
    p1.mostrar_datos()
    p1.vender(10)
    p1.reabastecer(30)
    p1.mostrar_datos()
